tests/test_decoding.py

```diff
@@ -590,12 +595,13 @@ def test_decode_request_structure(self):
             ecu_variant_patterns=[],
         )
         diag_layer = DiagLayer(diag_layer_raw=diag_layer_raw)
-        diag_layer.finalize_init(odxlinks=odxlinks)
+        diag_layer._resolve_odxlinks(odxlinks)
+        diag_layer._finalize_init(odxlinks)
 
-        req_param1._resolve_references(diag_layer, odxlinks)
-        req_param2._resolve_references(diag_layer, odxlinks)
-        struct_param1._resolve_references(diag_layer, odxlinks)
-        struct_param2._resolve_references(diag_layer, odxlinks)
+        req_param1._resolve_odxlinks(odxlinks)
+        req_param2._resolve_odxlinks(odxlinks)
+        struct_param1._resolve_odxlinks(odxlinks)
+        struct_param2._resolve_odxlinks(odxlinks)
```

Comment:
This is still required even after `diag_layer._resolve_odxlinks(odxlinks)` ?

Reply:
good catch. fixed: f8f1ab758c576d9319fe015aa8e4839a1d1ecf8d